internal/controller/runner_controller.go

```diff
@@ -0,0 +1,275 @@
+	if err != nil && garmClient.IsNotFoundError(err) {
+		return nil, nil
+	} else if err != nil {
+		return nil, err
+	}
+	return &garmRunner.Payload, err
+}
+
+func (r *RunnerReconciler) ensureFinalizer(ctx context.Context, runner *garmoperatorv1alpha1.Runner) error {
+	if !controllerutil.ContainsFinalizer(runner, key.RunnerFinalizerName) {
+		controllerutil.AddFinalizer(runner, key.RunnerFinalizerName)
+		return r.Update(ctx, runner)
+	}
+	return nil
+}
+
+func (r *RunnerReconciler) updateRunnerStatus(ctx context.Context, runner *garmoperatorv1alpha1.Runner, garmRunner *params.Instance) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+	log.Info("Update runner status...")
+
+	runner.Status.ID = garmRunner.ID
+	runner.Status.ProviderID = garmRunner.ProviderID
+	runner.Status.AgentID = garmRunner.AgentID
+	runner.Status.Name = garmRunner.Name
+	runner.Status.OSType = garmRunner.OSType
+	runner.Status.OSName = garmRunner.OSName
+	runner.Status.OSVersion = garmRunner.OSVersion
+	runner.Status.OSArch = garmRunner.OSArch
+	runner.Status.Addresses = garmRunner.Addresses
+	runner.Status.Status = garmRunner.Status
+	runner.Status.InstanceStatus = garmRunner.RunnerStatus
+	runner.Status.PoolID = garmRunner.PoolID
+	runner.Status.ProviderFault = garmRunner.ProviderFault
+	runner.Status.GitHubRunnerGroup = garmRunner.GitHubRunnerGroup
+
+	if err := r.Status().Update(ctx, runner); err != nil {
+		log.Error(err, "unable to update Runner status")
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *RunnerReconciler) SetupWithManager(mgr ctrl.Manager, eventChan chan event.GenericEvent) error {
+	c, err := ctrl.NewControllerManagedBy(mgr).
+		For(&garmoperatorv1alpha1.Runner{}).
+		Build(r)
+	if err != nil {
+		return err
+	}
+
+	return c.Watch(&source.Channel{Source: eventChan}, &handler.EnqueueRequestForObject{})
+}
+
+func (r *RunnerReconciler) PollRunnerInstances(ctx context.Context, eventChan chan event.GenericEvent) {
+	log := log.FromContext(ctx)
+	ticker := time.NewTicker(5 * time.Second)
+	for {
+		select {
+		case <-ctx.Done():
+			log.Info("Closing event channel for runners...")
+			close(eventChan)
+			return
+		case _ = <-ticker.C:
+			log.Info("Polling Runners...")
+			err := r.EnqueueRunnerInstances(ctx, eventChan)
+			if err != nil {
+				log.Error(err, "Failed polling runner instances")
+			}
+		}
+	}
+}
+
+func (r *RunnerReconciler) EnqueueRunnerInstances(ctx context.Context, eventChan chan event.GenericEvent) error {
+	allRunners := params.Instances{}
+	pools := &garmoperatorv1alpha1.PoolList{}
+
+	err := r.List(ctx, pools)
+	if err != nil {
+		return err
+	}
+
+	instanceClient := garmClient.NewInstanceClient()
+	err = instanceClient.Login(garmClient.GarmScopeParams{
+		BaseURL:  r.BaseURL,
+		Username: r.Username,
+		Password: r.Password,
+		// Debug:    true,
+	})
+	if err != nil {
+		return err
+	}
```

Comment:
We've intentionally decided to not "cache" the JWT token as we do not requesting the API-Server that often ([ADR](https://github.com/mercedes-benz/garm-operator/blob/main/docs/adrs/jwt_token_caching.md)).

But with this implementation i'm really struggling with that decision.
For this 5 second ticker we could think about the following for now:
* doing the login in `PollRunnerInstances` 
* passing the token to `EnqueueRunnerInstance`
  * If the JWT has expired, we might get a `401` and could catch that error
  * re-login and updating the JWT

At least we do not have to to the entire Login flow every 5 seconds